lib/fabrik/DBManager.js

```diff
@@ -42,86 +41,128 @@ const DB_STATES = {
-                .then(bindInfo => {
-                  this.dbCredentials = bindInfo.credentials;
-                  const connectParams = `username:${this.dbCredentials.username} - IPs @[10.11.252.20:27017,10.11.252.21:27017,10.11.252.22:27017]- DB : ${this.dbCredentials.dbname} - replicaSet : ${this.dbCredentials.replicaset}`;
-                  logger.info(`Bind Info retrieved - Connecting to : ${connectParams}`);
-                  return bindInfo.credentials.uri && this.initDb(_.assign({}, config.mongodb, {
-                    url: this.dbCredentials.uri
-                  }));
-                }).catch(ServiceBindingNotFound, (err) => logger.warn('MongoDB binding to ServiceFabrik not found. This generally should not occur. More Info:', err));
-              /**
-               *Could have automatically initiated create/update of DB deployment on start up when no binding/instance is found.
-               *However if broker goes HA then  on start only the master must provision / update the DB. Hence external hooks are
-               *provided for create / update, which can be suitably plugged in as part of post deployment hooks and targetted
-               *specfically to master node. At start up app only tries to bind an existing instance.
-               */
-            } else {
-              logger.error(`mongodb property - config.mongodb.deployment_name is undefined. Deployment name must be defined.`);
-              this.dbState = DB_STATES.NOT_CONFIGURED;
-            }
+      return utils
+        .retry(() => this
+          .directorManager
+          .getBindingProperty(config.mongodb.deployment_name, CONST.FABRIK_INTERNAL_MONGO_DB.BINDING_ID), {
+            maxAttempts: 5,
+            minDelay: 5000,
+            predicate: (err) => !(err instanceof ServiceBindingNotFound)
           });
-      } else if (config.mongodb.url) {
-        logger.info(`Connecting to DB with the provided config URL : ${config.mongodb.url}`);
-        this.initDb(config.mongodb);
-      } else {
-        logger.warn('Either DB URL or Mongo Plan Id must be configured for enabling DB usage with ServiceFabrik');
-        this.dbState = DB_STATES.NOT_CONFIGURED;
-      }
-    }).catch(err => {
-      logger.error('error occurred while initializing DBManager. More info:', err);
-      this.dbState = DB_STATES.INIT_FAILED;
     });
   }
 
   initDb(config) {
-    logger.info('Starting up db ...');
+    logger.info('Connecting to db ...');
     this.dbState = DB_STATES.CONNECTING;
     this.dbUrl = config.url;
-    return dbConnectionManager
-      .startUp(config)
-      .catch(() => this.dbState = DB_STATES.CONNECTION_FAILED);
+    return utils
+      .retry(() => dbConnectionManager
+        .startUp(config), {
+          maxAttempts: 5,
+          minDelay: 5000
+        })
+      .then(() => this.dbInitialized = true)
+      .catch((err) => {
```

Comment:
Do we need to log this error ?

Reply:
Error bubbles up and gets logged at one place centrally [here](https://github.com/jagadish-kb/service-fabrik-broker/blob/7aa3fda8089470c5e7f413fd66665fe097802b49/lib/fabrik/DBManager.js#L88)

Reply:
ok.